blockly-rtc/src/Position.js

```diff
@@ -37,14 +37,41 @@ export default class Position {
    * @public
    */  
   static fromEvent(event) {
-    // TODO: Add support for a field position on a change event.
-    const id = event.workspaceId;
+    if (event instanceof Blockly.Events.Ui) {
+      return this.fromUiEvent_(event);
+    } else if (event instanceof Blockly.Events.Change) {
+      return this.fromChangeEvent_(event);
+    } else {
+      throw Error('Cannot create position from this event.');
+    }
+  };
+
+  /**
+   * Create a Position from a Blockly UI event.
+   * @param {!Blockly.Event.Ui} event The event that creates a Position.
+   * @return {!PositionUpdate} The Position representative of the event.
+   * @private
+   */  
+  static fromUiEvent_(event) {
     const type = 'BLOCK';
     const blockId = event.newValue;
     const fieldName = null;
     return new Position(type, blockId, fieldName);  
   };
 
+  /**
+   * Create a Position from a Blockly Change event.
+   * @param {!Blockly.Event.Change} event The event that creates a Position.
+   * @return {!PositionUpdate} The Position representative of the event.
+   * @private
+   */  
+  static fromChangeEvent_(event) {
```

Comment:
What about non-field change events?

Reply:
Did more research into change events, now I made sure this function is only called on field change events and index.js is now only passing in field change events into this function.